src/main/java/com/leets/xcellentbe/domain/post/service/PostService.java

```diff
@@ -0,0 +1,58 @@
+package com.leets.xcellentbe.domain.post.service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.leets.xcellentbe.domain.post.domain.Post;
+import com.leets.xcellentbe.domain.post.domain.repository.PostRepository;
+import com.leets.xcellentbe.domain.post.dto.ArticlesResponseDto;
+import com.leets.xcellentbe.domain.user.domain.User;
+import com.leets.xcellentbe.domain.user.domain.repository.UserRepository;
+import com.leets.xcellentbe.domain.user.exception.UserNotFoundException;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class PostService {
+
+	private final UserRepository userRepository;
+	private final PostRepository postRepository;
+
+	// 게시글 조회 (미디어 필터 옵션 추가)
+	public List<ArticlesResponseDto> getArticles(String customId, boolean mediaOnly) {
+		User user = getUser(customId);
+		List<Object[]> posts = getPosts(user);
+
```

Comment:
`Object[]`를 사용한 이유가 궁금합니다!

Reply:
`getPosts` 메소드를 가보시면, `PostRepository`의 `findPostsByWriter`에서 값을 반환 받고 있습니다.

```java
	@Query("SELECT p, pm.filePath FROM Post p LEFT JOIN PostMedia pm ON p.postId = pm.post.postId WHERE p.writer = :user")
	List<Object[]> findPostsByWriter(User user);
```

`Post`와 `PostMedia`를 조인해서, `filePath`를 가져오다보니,, 마땅히 사용할만한 제네릭 타입이 없어 우선 `Object`로 지정해 두었습니다.

혹시 더 좋은 의견 있으시면 적극 반영하도록 하겠습니다 😭

Reply:
```
public record PostDto(
	Post post,
	String filePath
) {
}
```
이런식으로 따로 DTO를 생성해서

```suggestion
	@Query("SELECT new com.leets.xcellentbe.domain.post.dto.PostDto(p, pm.filePath) "
		+ "FROM Post p LEFT JOIN PostMedia pm ON p.postId = pm.post.postId " + "WHERE p.writer = :user")
	List<PostDto> findPostsByWriter(User user);
```
위와 같이 변경하시는 건 어떨까요?

Reply:
말씀하신 부분 반영하여 서비스 및 레포지토리 수정 하였습니다!